### DISCUSSION TITLE
[BUG] set-pnplistpermission group principal not found

### DISCUSSION THREAD

**Discussion by triga-s** [pnp/powershell#3951] [Q&A]
![image](https://github.com/pnp/powershell/assets/117661663/af8e1dc5-1e54-402e-9bb9-f7e84d950ffb)

As seen in the screenshot the group can be returned on get-pnpgroup. 
The $visitorgroup variable contains that group. 
I get the same error trying to call the group by $visitorgroup.title, $visitorgroup.id or anything else. 

Since pnp returns the group ID after calling it via Displayname or the object itself, it definitly is getting data from the group, but returns it can't find it anyway. 

Please help!

**Comment by markgort86** [comment]
This commandlet is just working fine in version 2.4.0 of PnP.PowerShell, first you need to break the role inheritance. See: https://pnp.github.io/powershell/cmdlets/Set-PnPList.html#-breakroleinheritance. Otherwise you get the following error: `Set-PnPListPermission: This operation is not allowed on an object that inherits permissions.`

![image](https://github.com/pnp/powershell/assets/52573368/9b4347c6-824a-4eb8-b33a-bcf341c20ce8)
If the visitors group have the following permissions and you just want to remove the read permissions and the contribute need to stay you can run the following cmdlet `Set-PnPListPermission -Identity Test -Group "TestSite Visitors" -RemoveRole Read`. Also if you want to remove the contribute, the only thing to do is change the value of the last parameter to "Contribute".

I could reproduce the error only when the visitors group is not assigned anymore to the list. Maybe this feedback helps to solve the problem.
